### DISCUSSION TITLE
How MOOSE computes eigen strain?

### DISCUSSION THREAD

**Discussion by galibubc** [idaholab/moose#23538] [Q&A Modules: Solid mechanics]
Hello,

From tensor mechanics module, it is possible to add eigenstrain to a linear elastic system (https://mooseframework.inl.gov/modules/tensor_mechanics/plug_n_play.html#qu2006fundamentals).

`ADComputeFiniteStrainElasticStress `can calculate stress due to finite strain including eigen strain.

Now $\varepsilon_{total} = \varepsilon_{eigen} + \varepsilon_{mechanical}$

`ADComputeFiniteStrainElasticStress` calculates  $\varepsilon_{mechanical}$ using Eulerian-Almansi finite strain tensor as $\varepsilon_{mechanical} = \frac{1}{2} (I - (\boldsymbol{F}.\boldsymbol{F}^{T})^{-1} )$.   **(is it correct?)**

How MOOSE calculates the total stress in this case? as hooks law is not valied for eigen strain so it couldn't be $\sigma = C. (\varepsilon_{eigen} + \varepsilon_{mechanical})$? Using green's function? how? Can you please explain the theory behind it?

Thank you.

**Comment by GiudGiud** [comment]
Hello

There's some documentation on eigenstrain here
https://mooseframework.inl.gov/moose/source/materials/ComputeEigenstrain.html
The references cited should explain the theory in more details

Guillaume

**Reply by galibubc** [reply]
Hello, 

Thanks. I already checked the references. Is it possible to briefly summarize how MOOSE deals with eigenstrain? I am confused how MOOSE handles finite strain and eigen strain at the same time to compute stress and strain energy. 

For instance using fourier integral-
![image](https://user-images.githubusercontent.com/121526655/221427215-9516052a-fca3-47a9-b79a-59137c079640.png)

Or Using Green function - 
![image](https://user-images.githubusercontent.com/121526655/221427356-b4496821-89e4-4612-bc4e-97b0a2ef34dd.png)

**Reply by GiudGiud** [reply]
@dschwen @recuero

**Reply by jiangwen84** [reply]
The tensor mechanics system you are looking at uses additive decomposition of the total incremental strain. 

The total incremental strain e_total = e_mechanical + e_eigenstrain and e_mechanical = e_elastic + e_plastic. 

The stress sigma = C * e_elastic.

**Reply by galibubc** [reply]
Hello,

Does that mean, in MOOSE, the nodal point displacement gives the deformation gradient. From where we can calculate the total strain. Then , elastic strain = total strain - inputed eigen strain. 

Then calculate the sigma = c * elastic strain ?
